### PR TITLE
Upgrade all dependencies to versions supporting Symfony 5

### LINKED ISSUE #11301: Support for JMS serializer 3.0
**Description**
To be able to install other libraries on my installation I want to have support for JMS Serializer 3.0, see https://github.com/schmittjoh/serializer/blob/master/UPGRADING.md#from-2x-to-300

Is it possible? Or are there issues mentioned as "deeper branch group exclusion strategy" has a different behaviour, the latest group is used instead of falling back to "Default" ?

I'm aware of https://github.com/Sylius/Sylius/issues/9288 but I think it'll take longer to implement and has a totally different path.

### PR DESCRIPTION
**Pull request by pamil**
Related to #10928, will make those records green:
- `friendsofsymfony/rest-bundle`
- `jms/serializer-bundle`
- `sylius/resource-bundle`
- `sylius/grid-bundle`
- `white-october/pagerfanta-bundle`
- `willdurand/hateoas`
- `willdurand/hateoas-bundle`
- `winzou/state-machine-bundle`
- `doctrine/doctrine-bundle`
- `doctrine/doctrine-cache-bundle`
- `friendsofsymfony/oauth-server-bundle`

Closes #11640, closes #11629, closes #11775, closes #11804, fixes #11301.

### CODE REVIEW

File: src/Sylius/Bundle/AdminApiBundle/composer.json [hunk 1]
```diff
@@ -24,15 +24,15 @@
     ],
     "require": {
         "php": "^7.3",
-        "friendsofsymfony/oauth-server-bundle": "^1.6",
+        "friendsofsymfony/oauth-server-bundle": "^1.6 || ^2.0@dev",
```
Comment (Zales0123) [inline src/Sylius/Bundle/AdminApiBundle/composer.json:27]: As far as I see, it breaks the build. The problem is, we have a spec for the `ClientManager` class that extends the `FOS\OAuthServerBundle\Entity\ClientManager`. In `^1.6` it requires `EntityManager` from `Doctrine\Common\Persistance`, but in `^2.0@dev` it wants it from new `Doctrine\Persistence` component. For some reason, the `1.6.2` version is installed in the build, not `2.0` - which results in an error.

It raises the question, can we even say this bundle is working with the `^1.6` version of this dependency? It was not a problem in the rest of the bundles, so should not be here either 🤷‍♂️

File: .github/workflows/application.yml [hunk 2]
```diff
@@ -366,7 +366,7 @@ jobs:
 
             -
                 name: Run JS Behat
-                run: vendor/bin/behat --colors --strict --no-interaction -vvv -f progress --tags="@javascript && ~@todo && ~@cli" || vendor/bin/behat --colors --strict --no-interaction -vvv -f progress --tags="@javascript && ~@todo && ~@cli" --rerun || vendor/bin/behat --colors --strict --no-interaction -vvv -f progress --tags="@javascript && ~@todo && ~@cli" --rerun
+                run: vendor/bin/behat --colors --strict --no-interaction -vvv --tags="@javascript && ~@todo && ~@cli" || vendor/bin/behat --colors --strict --no-interaction -vvv -f progress --tags="@javascript && ~@todo && ~@cli" --rerun || vendor/bin/behat --colors --strict --no-interaction -vvv -f progress --tags="@javascript && ~@todo && ~@cli" --rerun
```
[the remark this answers is not in the document]
Comment (pamil) [reply .github/workflows/application.yml:369]: It was used mostly in order to figure out which scenarios were causing Behat to time out - I'll revert it.

File: src/Sylius/Bundle/CoreBundle/EventListener/CircularDependencyBreakingErrorListener.php [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\Event\ControllerArgumentsEvent;
+use Symfony\Component\HttpKernel\Event\ExceptionEvent;
+use Symfony\Component\HttpKernel\EventListener\ErrorListener;
+use Symfony\Component\HttpKernel\EventListener\ErrorListener as SymfonyErrorListener;
+use Symfony\Component\HttpKernel\KernelEvents;
+
+/**
+ * @internal
```
[the remark this answers is not in the document]
Comment (pamil) [reply src/Sylius/Bundle/CoreBundle/EventListener/CircularDependencyBreakingErrorListener.php:16]: Indeed, an essay is incoming.

[the remark this answers is not in the document]
Comment (pamil) [reply src/Sylius/Bundle/CoreBundle/EventListener/CircularDependencyBreakingErrorListener.php:23]: @Zales0123 just added 🎉

File: src/Sylius/Bundle/AdminApiBundle/composer.json (hunk 1 above)
Comment (pamil) [reply src/Sylius/Bundle/AdminApiBundle/composer.json]: It is working with the current version, as well as with 2.0 - just the spec is something that we can't really fix, so I removed it, as it wasn't testing anything else than proxying a call in the end.

**Comment by loic425** [comment]
@pamil @Zales0123 Be careful, AdvancedUserInterface has been removed
User component needs symfony/security-core instead of symfony/security and maybe you need to introduce a pollyfill to prevent BC on this removed interface.

**Comment by Zales0123** [comment]
@loic425 good catch 👍 the solution is already proposed in https://github.com/Sylius/Sylius/pull/11774

**Comment by loic425** [comment]
@Zales0123 Is a PR already exist with the `Symfony\Component\EventDispatcher\Event` replacement to `Symfony\Contracts\EventDispatcher\Event`?

**Review by GSadee — APPROVED** [review]
It would be nice if the proper upgrade instructions were part of this PR

**Pull request merged**